src/compiler/checker.ts

```diff
@@ -8340,6 +8341,59 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
         }
 
+        function serializeTypeName(context: NodeBuilderContext, node: EntityName, isTypeOf?: boolean, typeArguments?: readonly TypeNode[]) {
+            const meaning = isTypeOf ? SymbolFlags.Value : SymbolFlags.Type;
+            const symbol = resolveEntityName(node, meaning, /*ignoreErrors*/ true);
+            if (!symbol) return undefined;

```

Comment:
The default behavior here is to fall back to serializing the type node when the name doesn't resolve, and usually I'm inclined to approve of that approach, but the `transpileDeclaration` issues we've been getting have made me keenly aware that *most* names won't resolve when transpiled under that mode, and you probably want the default behavior to be copy-the-node when the name doesn't resolve rather than print-any. This is somewhat reasonable since name-that-does-not-resolve is just `any` with more steps to get there.

Reply:
This actually isn't an issue the way this function is currently used. `trackExistingEntityName` will already not error if the symbol is not found and will just return the original name. `serializeTypeName` is always used as a fallback, when `trackExistingEntityName` fails so it will never be called if the symbol is not resolvable. 

I added a test that shows that the types are reused even if the symbol is not found.